pennylane/compiler/python_compiler/xdsl_extras/constraints.py

```diff
@@ -60,6 +60,9 @@ def mapping_type_vars(
         # pylint: disable=unused-argument,missing-function-docstring
         return self
 
+    def get_bases(self):
+        return set()
```

Comment:
This is _potentially_ a temporary change. There is ongoing discussion in the xDSL Zulip about the best way to handle the issue of setting a range of allowed ranks.